### PR TITLE
Port workspace history dialog to workbench

### LINKED ISSUE #24532: Port workpace history dialog to workbench
The context menu in the workspace list needs to show the "Show history" option.  The dialog is MantidPlot is fine so lets port this over and export it to Python.

### PR DESCRIPTION
**Pull request by ghost**
**Description of work.**
Port the WorkspaceHistoryWindow class used in MantidPlot to Workbench.

The "Show History" option will now be available when right-clicking on a workspace or table.

**Report to:** [nobody]. 

**To test:**
Run tests in "test_workspacewidget.py". 
Open workbench and load/create some workspaces/tables, right-click on them and select "Show History". Try selecting multiple workspaces and/or WorkspaceGroups.

Fixes #24532 . 

*This does not require release notes* because **workbench not yet released**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp [hunk 1]
```diff
@@ -101,13 +104,15 @@ void WorkspaceTreeWidgetSimple::popupContextMenu() {
       menu->addMenu(plotSubMenu);
       menu->addSeparator();
       menu->addAction(m_showData);
+      menu->addAction(m_showAlgorithmHistory);
```
Comment (martyngigg) [inline qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp]: I know you didn't add this option but while we're messing with the menu could we move the `Sample Logs` item from where it currently is to underneath "Show Instrument" and re-label it "Show Sample Logs"? It also only applies to `MatrixWorkspace` types so it doesn't need to be added to the items for a table.

**Review by martyngigg — CHANGES_REQUESTED** [review]
This looks good. Just a minor request for menu reordering.

**Review by martyngigg — COMMENTED** [review]
One final observation that is a little odd. With 2 monitors all of the other windows that are created with the "Show ..." options display themselves on top of the main application yet the history dialog always pops up on the left-hand side of the other monitor. Any idea why that might be?

**Comment by hsautessella** [comment]
I believe it's because AlgorithmHistoryWindow sets an absolute position through `setGeometry()` whereas, for example, the sample logs dialog doesn't explicitly set its position, it only calls `resize()`. 

The latest push should fix this. It's hard for me to tell how it will behave with multiple monitors, but from moving the window around on my screen, the AlgorithmHistoryWindow seems to always open in the centre of the workbench window now.

**Review by martyngigg — COMMENTED** [review]
This looks better, thanks. One more observation is when I select multiple workspaces and click "Show History" then the windows end up perfectly on top of each other so I can't tell that there are two open. If I do the same thing for the data or sample logs then the windows cascade so that you can see them. Could you look into this?

**Comment by martyngigg** [comment]
This is okay. The stacked windows is apparently default for the other widgets on Windows and it is not terribly high priority.

**Review by martyngigg — APPROVED** [review]

File: qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp (hunk 1 above)
Comment (rosswhitfield) [reply qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp]: @martyngigg :unamused: #25092

Comment (martyngigg) [reply qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp]: :man_facepalming: My bad. Sorry!

File: qt/applications/workbench/workbench/plugins/workspacewidget.py [hunk 2]
```diff
@@ -137,7 +139,19 @@ def _do_show_data(self, names):
...
                     logger.error(
-                        "Could not open workspace: {0} with neither MatrixWorkspaceDisplay nor TableWorkspaceDisplay.")
+                        "Could not open workspace: {0} with neither "
+                        "MatrixWorkspaceDisplay nor TableWorkspaceDisplay."
+                        "".format(ws.name()))
+
+    def _do_show_algorithm_history(self, names):
+        for name in names:
+            if any(isinstance(self._ads.retrieve(name), ws_type)
+                   for ws_type in [MatrixWorkspace, ITableWorkspace]):
```
Comment (rosswhitfield) [inline qt/applications/workbench/workbench/plugins/workspacewidget.py]: @hsautessella @martyngigg Why was MDWorkspaces excluded?

Comment (hsautessella) [reply qt/applications/workbench/workbench/plugins/workspacewidget.py]: I've opened a PR #25119. Do let me know if there are any other options that may need adding.

**Pull request merged**